### ISSUE TITLE
Can't create new content types after migrating to 3.0.0-alpha.25.2

### ISSUE THREAD

**Issue by selfagency** [strapi/strapi#3123]
**Informations**
- **Node.js version**: 10.15.3
- **NPM version**: 6.9.0
- **Strapi version**: 3.0.0-alpha.24.1
- **Database**: Postgres
- **Operating system**:  Ubuntu 18.04
- **(Optional) Link to your Project**: https://gitlab.com/selfagencyllc/kifa-cms

**What is the current behavior?**
After updating to 3.0.0-alpha.25.2, when attempting to add a new content type, the type saves in the API folder, but nothing happens in the database to create new tables nor save the schema. If I delete and retry adding the type, I get a `request.error.model.exist` error.

**Steps to reproduce the problem**
Download my project and see for yourself?

**What is the expected behavior?**
The DB should be updated to add the new tables and save the schema.

**Suggested solutions**
🤷‍♂️

**Comment by derrickmehaffy** [comment]
@selfagency delete the `plugin_content-manager_schema` entry in your `core_store` table.

**Comment by selfagency** [comment]
I did that, and now I'm getting—

```
Unhandled promise rejection TypeError: "n.models is null"
    s main.js:48
main.js:6
    t main.js:6
    exports main.js:6
    I main.js:6
    exports main.js:6
    <anonymous> main.js:6
    v main.js:6
    _ main.js:6
TypeError: "e.get(...) is null"
    h main.js:48
    c main.js:72
    i main.js:72
    d main.js:72
    i main.js:72
    d main.js:72
    i main.js:72
    r main.js:72
    l main.js:72
    u main.js:72
    o main.js:72
    run main.js:72
    onStateChange main.js:72
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    l main.js:45
    n main.js:37
    e main.js:45
    dispatch main.js:45
    F main.js:45
    x main.js:45
    n main.js:45
    s main.js:45
    r main.js:45
    x main.js:45
    b main.js:45
    h main.js:45
    o main.js:45
main.js:16
uncaught at o Error: e.get(...) is null
h/<@http://localhost:1337/admin/content-manager/main.js:48:504996
a/</c<@http://localhost:1337/admin/content-manager/main.js:72:98952
i/<@http://localhost:1337/admin/content-manager/main.js:72:98379
a/</d<@http://localhost:1337/admin/content-manager/main.js:72:99082
i/<@http://localhost:1337/admin/content-manager/main.js:72:98379
a/</d<@http://localhost:1337/admin/content-manager/main.js:72:99047
i/<@http://localhost:1337/admin/content-manager/main.js:72:98379
i/</r<@http://localhost:1337/admin/content-manager/main.js:72:24433
l@http://localhost:1337/admin/content-manager/main.js:72:22362
u@http://localhost:1337/admin/content-manager/main.js:72:22480
o/<@http://localhost:1337/admin/content-manager/main.js:72:22598
run@http://localhost:1337/admin/content-manager/main.js:72:12377
i/</c</d.onStateChange@http://localhost:1337/admin/content-manager/main.js:72:15241
notify@http://localhost:1337/admin/main.js:37:44740
./node_modules/strapi-helper-plugin/node_modules/react-redux/lib/utils/Subscription.js/i</t.notifyNestedSubs@http://localhost:1337/admin/main.js:37:45242
a/</d</p.onStateChange@http://localhost:1337/admin/main.js:37:33488
notify@http://localhost:1337/admin/main.js:37:44740
./node_modules/strapi-helper-plugin/node_modules/react-redux/lib/utils/Subscription.js/i</t.notifyNestedSubs@http://localhost:1337/admin/main.js:37:45242
a/</d</p.onStateChange@http://localhost:1337/admin/main.js:37:33488
notify@http://localhost:1337/admin/main.js:37:44740
./node_modules/strapi-helper-plugin/node_modules/react-redux/lib/utils/Subscription.js/i</t.notifyNestedSubs@http://localhost:1337/admin/main.js:37:45242
a/</d</p.onStateChange@http://localhost:1337/admin/main.js:37:33488
notify@http://localhost:1337/admin/main.js:37:44740
./node_modules/strapi-helper-plugin/node_modules/react-redux/lib/utils/Subscription.js/i</t.notifyNestedSubs@http://localhost:1337/admin/main.js:37:45242
a/</d</p.onStateChange@http://localhost:1337/admin/main.js:37:33488
notify@http://localhost:1337/admin/main.js:37:44740
./node_modules/strapi-helper-plugin/node_modules/react-redux/lib/utils/Subscription.js/i</t.notifyNestedSubs@http://localhost:1337/admin/main.js:37:45242
a/</d</p.onStateChange@http://localhost:1337/admin/main.js:37:33488
l@http://localhost:1337/admin/main.js:45:127219
n/</</<@http://localhost:1337/admin/main.js:37:67684
e/</<@http://localhost:1337/admin/main.js:45:103757
dispatch@http://localhost:1337/admin/main.js:45:131259
F/<@http://localhost:1337/admin/main.js:45:124339
x/<@http://localhost:1337/admin/main.js:45:108413
n@http://localhost:1337/admin/main.js:45:119621
s@http://localhost:1337/admin/main.js:45:119771
r@http://localhost:1337/admin/main.js:45:119670
x@http://localhost:1337/admin/main.js:45:108370
b@http://localhost:1337/admin/main.js:45:107576
h@http://localhost:1337/admin/main.js:45:106515
o@http://localhost:1337/admin/main.js:45:107123
```

—and this is my localhost and *not* behind Nginx or any other proxy.

**Comment by selfagency** [comment]
Wait, I see I needed to delete the whole row and not just the contents. Which is now done and so the above error is no longer a problem. But it still didn't create the new database tables:

```
insert into `statements` (`created_at`, `published`, `title`, `updated_at`, `url_slug`) values ('2019-04-13 11:32:56.683', true, 'Test', '2019-04-13 11:32:56.683', 'test') - ER_NO_SUCH_TABLE: Table 'kifa_strapi.statements' doesn't exist
```

**Comment by derrickmehaffy** [comment]
@selfagency you have `_` in some of your field names: https://gitlab.com/selfagencyllc/kifa-cms/blob/master/api/campaigns/models/Campaigns.settings.json

Symbols aren't supported in model or field names. On **non-relation** fields you can kinda get away with camelCase names (though also not officially supported) but symbols will break stuff.

https://github.com/strapi/strapi/wiki/Frequently-Asked-Questions#model--field-naming-schemes

This is a known problem that will be looked in the future but was deemed a low priority issue for now.

**Comment by selfagency** [comment]
It worked fine until I tried to add these new content types. Are you sure this is why it can't create the new tables?

**Comment by derrickmehaffy** [comment]
@selfagency It is highly likely yes, you are jumping from v24.1 to v25.2 which is quite a few versions, depending on how you attempted to update your project there could be a lot of root causes but since symbols are not supported in field names at all, this is a good first step to finding the root cause.

I would ask that you modify all your models to remove the symbols for now, make a backup of your database, delete the database, and try again.

**Comment by selfagency** [comment]
If Strapi doesn't support symbols, why is it that the API autogenerates fields called `created_at` and `updated_at` and `public_id`?

**Comment by derrickmehaffy** [comment]
@selfagency they are hard coded

**Comment by selfagency** [comment]
So I migrated my fields to remove the underscores, and I'm getting the same exact problem. 

```
0|kifa-cms  | { Error: ER_NO_SUCH_TABLE: Table 'kifa_strapi.statements' doesn't exist
0|kifa-cms  |     at Query.Sequence._packetToError (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
0|kifa-cms  |     at Query.ErrorPacket (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/sequences/Query.js:77:18)
0|kifa-cms  |     at Protocol._parsePacket (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/Protocol.js:278:23)
0|kifa-cms  |     at Parser.write (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/Parser.js:76:12)
0|kifa-cms  |     at Protocol.write (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/Protocol.js:38:16)
0|kifa-cms  |     at Socket.<anonymous> (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/Connection.js:91:28)
0|kifa-cms  |     at Socket.<anonymous> (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/Connection.js:502:10)
0|kifa-cms  |     at Socket.emit (events.js:197:13)
0|kifa-cms  |     at Socket.EventEmitter.emit (domain.js:446:20)
0|kifa-cms  |     at addChunk (_stream_readable.js:288:12)
0|kifa-cms  |     at readableAddChunk (_stream_readable.js:269:11)
0|kifa-cms  |     at Socket.Readable.push (_stream_readable.js:224:10)
0|kifa-cms  |     at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
0|kifa-cms  |     at Protocol._enqueue (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/Protocol.js:144:48)
0|kifa-cms  |     at Connection.query (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/Connection.js:200:25)
0|kifa-cms  |     at /Users/daniel/code/kifa-cms/node_modules/knex/lib/dialects/mysql/index.js:144:18
0|kifa-cms  | From previous event:
0|kifa-cms  |     at Client_MySQL._query (/Users/daniel/code/kifa-cms/node_modules/knex/lib/dialects/mysql/index.js:135:12)
0|kifa-cms  |     at Client_MySQL.query (/Users/daniel/code/kifa-cms/node_modules/knex/lib/client.js:192:17)
0|kifa-cms  |     at Runner.<anonymous> (/Users/daniel/code/kifa-cms/node_modules/knex/lib/runner.js:138:36)
0|kifa-cms  | From previous event:
0|kifa-cms  |     at /Users/daniel/code/kifa-cms/node_modules/knex/lib/runner.js:47:21
0|kifa-cms  | From previous event:
0|kifa-cms  |     at Runner.run (/Users/daniel/code/kifa-cms/node_modules/knex/lib/runner.js:33:30)
0|kifa-cms  |     at Builder.Target.then (/Users/daniel/code/kifa-cms/node_modules/knex/lib/interface.js:23:43)
0|kifa-cms  |     at processImmediate (timers.js:637:19)
0|kifa-cms  |     at process.topLevelDomainCallback (domain.js:126:23)
0|kifa-cms  |   code: 'ER_NO_SUCH_TABLE',
0|kifa-cms  |   errno: 1146,
0|kifa-cms  |   sqlMessage: "Table 'kifa_strapi.statements' doesn't exist",
0|kifa-cms  |   sqlState: '42S02',
0|kifa-cms  |   index: 0,
0|kifa-cms  |   sql: 'select count(*) as `count` from `statements`' }
```

When I tried deleting my database all together and starting from scratch and it gave me:
```
error { Error: ER_NO_SUCH_TABLE: Table 'kifa_strapi.core_store' doesn't exist
    at Query.Sequence._packetToError (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/sequences/Query.js:77:18)
    at Protocol._parsePacket (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/Protocol.js:278:23)
    at Parser.write (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/Parser.js:76:12)
    at Protocol.write (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/Connection.js:91:28)
    at Socket.<anonymous> (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/Connection.js:502:10)
    at Socket.emit (events.js:197:13)
    at Socket.EventEmitter.emit (domain.js:446:20)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
    at Protocol._enqueue (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at Connection.query (/Users/daniel/code/kifa-cms/node_modules/mysql/lib/Connection.js:200:25)
    at /Users/daniel/code/kifa-cms/node_modules/knex/lib/dialects/mysql/index.js:144:18
From previous event:
    at Client_MySQL._query (/Users/daniel/code/kifa-cms/node_modules/knex/lib/dialects/mysql/index.js:135:12)
    at Client_MySQL.query (/Users/daniel/code/kifa-cms/node_modules/knex/lib/client.js:192:17)
    at Runner.<anonymous> (/Users/daniel/code/kifa-cms/node_modules/knex/lib/runner.js:138:36)
From previous event:
    at /Users/daniel/code/kifa-cms/node_modules/knex/lib/runner.js:47:21
From previous event:
    at Runner.run (/Users/daniel/code/kifa-cms/node_modules/knex/lib/runner.js:33:30)
    at Builder.Target.then (/Users/daniel/code/kifa-cms/node_modules/knex/lib/interface.js:23:43)
    at processImmediate (timers.js:637:19)
    at process.topLevelDomainCallback (domain.js:126:23)
  code: 'ER_NO_SUCH_TABLE',
  errno: 1146,
  sqlMessage: "Table 'kifa_strapi.core_store' doesn't exist",
  sqlState: '42S02',
  index: 0,
  sql:
   "select `core_store`.* from `core_store` where `core_store`.`key` = 'plugin_content-manager_schema' and `core_store`.`environment` = '' and `core_store`.`tag` = '' limit 1" }
```

**Comment by selfagency** [comment]
Please if you have any ideas, I'm overdue on this project and I keep getting held up on stupid snags like this.

**Comment by selfagency** [comment]
My client just texted me this.
![IMG_4001](https://user-images.githubusercontent.com/2541728/56094380-c5a23b80-5ea1-11e9-8f85-1abded1aa4d9.gif)

**Comment by derrickmehaffy** [comment]
@selfagency Jump on the slack, I will grab your source and try to load it myself.
https://slack.strapi.io

**Comment by derrickmehaffy** [comment]
Found the problem, in your database.json you were missing the `"options": {}` which is what caused the error.

```
{
  "defaultConnection": "default",
  "connections": {
    "default": {
      "connector": "strapi-hook-bookshelf",
      "settings": {
        "client": "mysql",
        "host": "${process.env.DB_HOST}",
        "port": "${process.env.DB_PORT}",
        "database": "${process.env.DB_NAME}",
        "username": "${process.env.DB_USER}",
        "password": "${process.env.DB_PASS}",
        "timezone": "utc"
      },
      "options": {}
    }
  }
}
```

**Comment by derrickmehaffy** [comment]
Per slack @selfagency stated the problem was fixed by the above ^^ closing

**Comment by selfagency** [comment]
Thank you very much for your help.

**Comment by selfagency** [comment]
@derrickmehaffy Looks like I spoke too soon. After porting the updated code to my server and migrating the database, I'm getting this when I try to add new entries to the new types I created:

<img width="1177" alt="Screenshot 2019-04-14 22 47 35" src="https://user-images.githubusercontent.com/2541728/56104885-a3db9f80-5f07-11e9-8a09-cd99b27636ff.png">

**Comment by selfagency** [comment]
Ah I deleted the content manager plugin schema again and that did it.

**Comment by derrickmehaffy** [comment]
:wink: Hopefully that annoying bug will be fixed soon with the CTM rewrite
